src/ansys/geometry/core/math/vector.py

```diff
@@ -265,6 +266,21 @@ class representing the second point.
         """
         return Vector3D(point_b - point_a)
 
+    @staticmethod
+    def rotate_vector(_this: "Vector3D", vector: "Vector3D", angle: float) -> "Vector3D":
+        """Vector3D helper method -- rotate_vector."""
+        if _this.is_zero:
+            raise Exception("Invalid vector operation.")
+
+        angle_between = vector.get_angle_between(_this)
+        parallel = Vector3D(
+            vector.x * angle_between, vector.y * angle_between, vector.z * angle_between
+        )
+
+        perpendicular1 = vector - parallel
+        perpendicular2 = _this.cross(perpendicular1)
+        return parallel + perpendicular1 * math.cos(angle) + perpendicular2 * math.sin(angle)
```

Comment:
Watch out... you are performing many assumptions here.

1) Why are you multiplying the coordinates against the angle alone? If you are assuming an infinitesimal angle (i.e. very small) and you are approximating the sine to the angle value... you could have a point. But you cannot assume the angle is small by default. Please revisit.
2) What's the reference for this algorithm? Is there some text-book where it is showing up? I can't replicate your math sadly. Please share the link. Also, we need to improve the in-line comments and explanations

Reply:
@RobPasMue makes sense, let me rework on it

Reply:
Sure! Feel free to reach out if you want to discuss anything =)